src/main/java/com/google/devtools/build/lib/runtime/QueryRuntimeHelper.java

```diff
@@ -63,8 +68,20 @@ class StdoutQueryRuntimeHelperFactory implements Factory {
     private StdoutQueryRuntimeHelperFactory() {}
 
     @Override
-    public QueryRuntimeHelper create(CommandEnvironment env) {
-      return createInternal(env.getReporter().getOutErr().getOutputStream());
+    public QueryRuntimeHelper create(CommandEnvironment env, CommonQueryOptions options)
+        throws QueryRuntimeHelperException {
+      PathFragment outputFile = options.outputFile;
```

Comment:
You should be able to access the query options from `env` here, and avoid changing the factory interface:

```
var queryOptions = env.getOptions().getOptions(CommonQueryOptions.class);
```

Reply:
That doesn't work, I guess because `CommonQueryOptions` is subclassed by the actual query options class ([example](https://github.com/bazelbuild/bazel/blob/74096f6757d717d5bb2dc6430fc67a2c93bf1fea/src/main/java/com/google/devtools/build/lib/query2/cquery/CqueryOptions.java#L25)) and thus does not itself exist in the options map.

Reply:
`QueryOptions`?

Reply:
It was easy to make it work for `query`, `cquery`, and `aquery`, so I did.